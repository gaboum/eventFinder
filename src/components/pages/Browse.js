/**
 * Created by HP on 23-Dec-17.
 */
import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types'
import {CSSTransitionGroup} from 'react-transition-group';

import FiltersForm from '../FiltersForm';
import FiltersBar from '../FiltersBar';
import Event from '../Event';
import {getFilteredEvents} from '../../actions/events';
import {getUsersLocality} from '../../actions/userData';
import {filtersAreSame} from '../../helpers/compareFilters';
import Loading from '../loading';




/**
 * Represents the Browse page
 */
class Browse extends React.Component {

    /**
     * Starts an AJAX request to obtain filtered events in accordance with user's input
     * Gets events baised on user's location on initial loac
     */
    componentDidMount() {
        this.props.getFilteredEvents(this.props);
    }


    /**
     * Gets filtered events accordingly with user's input
     * Dispatches a new AJAX request if filters were changed
     * @param nextProps
     */
    componentDidUpdate(nextProps){
        this.props.getUsersLocality(nextProps.userData.location);
        if (!filtersAreSame(nextProps.filters, this.props.filters)) {
            this.props.getFilteredEvents(this.props);
        }
    }


    render(){
        return(
            <div className="browse-page">
                <div className="container-fluid">
                    <div className="row">
                        <div className="col-md-3">
                            <FiltersForm/>
                        </div>
                        <div className="col-md-9">
                            <FiltersBar/>
                            {this.props.events.length
                                ?
                                <CSSTransitionGroup
                                    transitionName="mount-animation"
                                    transitionAppear={true}
                                    transitionAppearTimeout={500}
                                    transitionEnter={false}
                                    transitionLeave={false}>
                                    {this.props.events.map(ev => {
                                        return <Event
                                            key={ev.id}
                                            event={ev}
                                        />
                                    })}
                                </CSSTransitionGroup>
                                :
                                <Loading/>
                            }
                        </div>
                    </div>
                </div>

            </div>
        )
    }
}


/**
 * Validation of props
 * @type {{userData, filters}}
 */
Browse.propTypes = {
    userData : PropTypes.object.isRequired,
    filters  : PropTypes.object.isRequired,
};


/**
 * Mapping filters and user's data to props
 * @param state
 */
const mapStateToProps = state => ({
    filters      : state.filters,
    userData     : state.userData,
    events       : state.events.filteredEvents
});


/**
 * Mapping dispatching of filtered events AJAX and user locality AJAX to google API
 * @param dispatch
 */
const mapDispatchToProps = dispatch => ({
    getFilteredEvents : args => dispatch(getFilteredEvents(args)),
    getUsersLocality  : args => dispatch(getUsersLocality(args))
});

export default connect(mapStateToProps, mapDispatchToProps)(Browse);


