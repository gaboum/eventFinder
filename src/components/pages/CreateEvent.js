/**
 * Created by HP on 23-Dec-17.
 */
import React from 'react';
import {connect} from 'react-redux';
import {Field, reduxForm} from 'redux-form';
import moment from 'moment';
import { Editor } from '@tinymce/tinymce-react';
import { Input } from 'antd';

import DateTimeSet from '../DateTimeSet';
import FileUpload from '../FileUpload';
import Validator from '../../helpers/fieldValidator';



class CreateEvent extends React.Component {

    render(){
        return (
            <div className="create-event-container">
                <form className="create-event__form" onSubmit={this.submit}>
                    <div className="create-event__fieldset">
                        <label htmlFor="title">Event Title</label>
                        <Field name="title" component={Input} type="text" />
                    </div>
                    <div className="create-event__fieldset">
                        <label htmlFor="location">Location</label>
                        <Field name="location" component={Input} type="text" />
                    </div>
                    <div className="create-event__fieldset">
                        <label htmlFor="start-date">Start Date</label>
                        <Field
                            name="start-date"
                            component={DateTimeSet}
                            context="today"
                            defaultDate={moment()}
                            defaultTime={moment()}
                            type="text"
                        />
                    </div>
                    <div className="create-event__fieldset">
                        <label htmlFor="start-date">End Date</label>
                        <Field
                            name="end-date"
                            component={DateTimeSet}
                            context="today"
                            defaultDate={moment()}
                            defaultTime={moment()}
                            type="text"
                        />
                    </div>
                    <div className="create-event__fieldset">
                        <Field name="picture" component={FileUpload}/>
                    </div>
                    <div className="create-event__fieldset">
                        <Field name="picture" component={Editor}
                               initialValue="<p>This is the initial content of the editor</p>"
                               init={{
                                   plugins: 'link image code',
                                   toolbar: 'undo redo | bold italic | alignleft aligncenter alignright | code'
                               }}/>
                    </div>
                    <div className="create-event__fieldset">
                        <label htmlFor="title">Organizer Name</label>
                        <Field name="organizer_name" component={Input} type="text" />
                    </div>

                </form>
            </div>
        )
    }
}

const validate = values => {
    const errors = {};
    if(!Validator.isRequired(values.title)){
        errors.title = 'Title is required';
    }
};

const connectedForm = connect()(CreateEvent);

export default reduxForm({
    form: 'create'
})(connectedForm);