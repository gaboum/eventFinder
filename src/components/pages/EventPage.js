/**
 * Created by HP on 04-Jan-18.
 */
import React from 'react';
import {connect} from 'react-redux';
import moment from 'moment';

import {getVenue, resetVenue} from '../../actions/events';
import {findEventById} from '../../selectors/events';


/**
 * Represent the Event page
 * @param props
 * @returns {XML}
 * @constructor
 */
class EventPage extends React.Component {


    /**
     * Fulfils an AJAX request to get venue's info
     */
    componentDidMount() {
        this.props.setVenue(this.props.event[0].venue_id);
    }



    /**
     * Sets current venue to an empty object when uset leaves the page
     */
    componentWillUnmount() {
        this.props.resetVenue();
    }



    render() {
        const { logo, name, is_free, description, start, end} = this.props.event[0];
        const {address_1, address_2, city, country, name : venue_name} = this.props.venue;

        return (
            <div className="event-page__background">
                <div className="event-page__info">
                    <div className="event-page__header">
                        <div className="event-page__picture-holder">
                            <img src={logo && logo.url} />
                        </div>
                        <div className="event-page__basic-info">
                            <div className="event-page__start">
                                {start && moment(start.local).format('MMM Do')}
                            </div>
                            <div className="event-page__item-name">{name && name.text}</div>
                            {is_free && <div className="event__price-types">FREE</div>}
                        </div>
                    </div>
                    <hr className="divider"/>
                    <div className="event-page__body">
                        <div className="event-page__description"><span className="event-page__description">Description: </span>
                            {description.text ? description.text : ''}
                        </div>
                        <div className="event-page__venue">
                            <div className="event-page__venue-element">{address_1 && <p>{address_1}</p>}</div>
                            <div className="event-page__venue-element">{address_2 && <p>{address_2}</p>}</div>
                            <div className="event-page__venue-element">{city && <p>{city}</p>}</div>
                            <div className="event-page__venue-element">{country && <p>{country}</p>}</div>
                            <div className="event-page__venue-element">{venue_name && <p>{venue_name}</p>}</div>
                            <div className="event--page__date">
                                {start && moment(start.local).format('DD')} - {end && moment(end.local).format('DD MMM')}
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}


const mapStateToProps = (state, props) => ({
    event : findEventById(state, props.match.params.id),
    venue : state.events.currentEventVenue
});



const mapDispatchToProps = dispatch => ({
    setVenue : id => dispatch(getVenue(id)),
    resetVenue : () => dispatch(resetVenue()),
});


export default connect(mapStateToProps, mapDispatchToProps)(EventPage);