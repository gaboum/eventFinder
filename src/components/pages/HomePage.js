/**
 * Created by HP on 23-Dec-17.
 */
import React from 'react';
import {connect} from 'react-redux';

import {getEvents, setError} from '../../actions/events';


class HomePage extends React.Component {
    componentWillMount() {
        const getLocation = () => {
            const geolocation = navigator.geolocation;

            return  new Promise((resolve, reject) => {
                if (!geolocation) {
                    reject(new Error('Not Supported'));
                }

                geolocation.getCurrentPosition((position) => {
                    //console.log(position.coords);
                    resolve(position);
                }, () => {
                    reject(new Error('Permission denied'));
                });
            });
        };

        getLocation()
            .then(location => this.props.getEvents(location.coords));


    }


    render(){
        return(
            <h1>HomePage</h1>
        )
    }
}

const mapDispatchToProps = dispatch => ({
    getEvents : (cords) => dispatch(getEvents(cords)),
    setError : (error) => dispatch(setError(error))
});

export default connect(undefined, mapDispatchToProps)(HomePage);