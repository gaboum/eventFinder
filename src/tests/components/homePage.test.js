/**
 * Created by HP on 08-Jan-18.
 */
import React from 'react';
import { shallow} from 'enzyme';
import {HomePage} from '../../components/pages/HomePage';

let wrapper, getEvents, getLocality;

beforeEach(() => {
    getEvents = jest.fn();
    getLocality = jest.fn();
    wrapper = shallow(<HomePage
        getEvents={getEvents}
        getLocality={getLocality}
    />)
});

test('should render the home page properly', () => {
    expect(wrapper).toMatchSnapshot();
});

/*test('should call getEvents and getLocality functions on component update', () => {

});*/