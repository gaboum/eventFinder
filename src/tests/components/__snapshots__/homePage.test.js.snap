// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the home page properly 1`] = `undefined`;
